core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -927,27 +928,43 @@ private void visitAllEntities(Solution_ solution, Consumer<Object> visitor,
         }
     }
 
-    public List<Object> getEntityListByEntityClass(Solution_ solution, Class<?> entityClass) {
-        List<Object> entityList = new ArrayList<>();
+    public void visitEntitiesByEntityClass(Solution_ solution, Class<?> entityClass, Consumer<Object> visitor) {
         for (MemberAccessor entityMemberAccessor : entityMemberAccessorMap.values()) {
-            if (entityMemberAccessor.getType().isAssignableFrom(entityClass)) {
+            if (entityClass.isAssignableFrom(entityMemberAccessor.getType())) {
                 Object entity = extractMemberObject(entityMemberAccessor, solution);
-                if (entity != null && entityClass.isInstance(entity)) {
-                    entityList.add(entity);
+                if (entity != null) {
+                    visitor.accept(entity);
                 }
             }
         }
         for (MemberAccessor entityCollectionMemberAccessor : entityCollectionMemberAccessorMap.values()) {
-            // TODO if (entityCollectionPropertyAccessor.getPropertyType().getElementType().isAssignableFrom(entityClass)) {
-            Collection<Object> entityCollection = extractMemberCollectionOrArray(entityCollectionMemberAccessor, solution,
-                    false);
-            for (Object entity : entityCollection) {
-                if (entityClass.isInstance(entity)) {
-                    entityList.add(entity);
+            Optional<Class<?>> optionalTypeParameter = ConfigUtils.extractCollectionGenericTypeParameterLeniently(
+                    "solutionClass", entityCollectionMemberAccessor.getDeclaringClass(),
+                    entityCollectionMemberAccessor.getType(),
+                    entityCollectionMemberAccessor.getGenericType(),
+                    null,
+                    entityCollectionMemberAccessor.getName());
+            if (optionalTypeParameter.isPresent()) {
```

Comment:
Why not [`ifPresentOrElse(...)`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Optional.html#ifPresentOrElse(java.util.function.Consumer,java.lang.Runnable))?

Reply:
Because apparently it's not on Android:
```
INFO] --- animal-sniffer-maven-plugin:1.20:check (animal-sniffer) @ optaplanner-core-impl ---
[INFO] Checking unresolved references to net.sf.androidscents.signature:android-api-level-32:12_r1
[ERROR] /mnt/fonsadae/src/github.com/kiegroup/optaplanner/core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/solution/descriptor/SolutionDescriptor.java:947: Undefined reference: void java.util.Optional.ifPresentOrElse(java.util.function.Consumer, Runnable)
```